spec/models/ability_spec.rb

```diff
@@ -1,437 +1,240 @@
-            it 'allows to see not hidden email address' do
-              other_user.hide_email = false
-              expect(ability).to be_able_to(:read_email, other_user)
-            end
-          end
-
-          context 'when a confirmed user' do
-            before do
-              allow(user).to receive(:admin?).and_return(false)
-              allow(user).to receive(:confirmed?).and_return(true)
-            end
-            it 'allows to see not hidden email address' do
-              other_user.hide_email = false
-              expect(ability).to be_able_to(:read_email, other_user)
-            end
-          end
-
-          context 'when the user him/herself' do
-            it 'allows to see not hidden email address' do
-              user.hide_email = false
-              expect(ability).to be_able_to(:read_email, user)
-            end
-          end
+      describe "Student" do
+        let(:student_team) { build(:team, :in_current_season) }
+        let(:second_team) { build(:team, :in_current_season) }
+        before do
+          allow(user).to receive(:current_student?).and_return(true)
         end
+        # TODO Question: is this ^ stub correct to test the ability only?
 
-      end
-
-      describe 'who is disallowed to see email address in user profile' do
+        it_behaves_like "same public features as confirmed user"
+        it { expect(subject).to be_able_to(:create, Conference) }
 
-        # email address is hidden: not admin, not user's supervisor in current season
-        # email address is not hidden: not admin, not confirmed user, not user him/herself
-
-        let(:other_user) { create(:user) }
-
-        context 'when email address is hidden' do
-          context "when not an admin or user's supervisor in current season" do
-            before do
-              allow(user).to receive(:admin?).and_return(false)
-              allow(ability).to receive(:supervises?).with(other_user, user).and_return(false)
-            end
-            it 'disallows to see hidden email address' do
-              other_user.hide_email = true
-              expect(ability).not_to be_able_to(:read_email, other_user)
-            end
-          end
+        it "can create their first team" do
+          expect(subject).to be_able_to(:create, student_team)
         end
-
-        context 'when email address is not hidden' do
-          context 'when not an admin or confirmed user or user him/herself' do
-            before do
-              allow(user).to receive(:admin?).and_return(false)
-              allow(user).to receive(:confirmed?).and_return(false)
-            end
-            it 'disallows to see not hidden email address' do
-              other_user.hide_email = false
-              expect(ability).not_to be_able_to(:read_email, other_user)
-            end
-          end
+        it 'cannot create a second team in current_season' do
+          pending 'fails; it is tested in team_spec.rb:33, and that passes'
```

Comment:
Well, I fear you have to actually write the teams to the DB to have a test like this...

Reply:
That ^ and we have this for confirmed users on line 27:

```ruby
can [:join, :create], Team
```

By the way, instead of persisting the teams in the DB maybe you can get away with stubbing the `user.teams` call.